processor/k8sattributesprocessor/pod_association.go

```diff
@@ -91,6 +91,14 @@ func getConnectionIP(ctx context.Context) kube.PodIdentifier {
 	if c.Addr == nil {
 		return ""
 	}
+	switch addr := c.Addr.(type) {
+	case *net.UDPAddr:
+		return kube.PodIdentifier(addr.IP.String())
+	case *net.TCPAddr:
+		return kube.PodIdentifier(addr.IP.String())
+	case *net.IPAddr:
+		return kube.PodIdentifier(addr.IP.String())
```

Comment:
Can we collapse all the cases?
```
case *net.UDPAddr, *net.TCPAddr, *net.IPAddr:
```

Reply:
We can't - the Go type switch would keep `addr` as a `net.Addr` type in this case, which does not have the IP property on it.

Reply:
Got it. thanks for explanation